### PR TITLE
feat(load_plugins): ignore hexo-theme-[config.theme]

### PR DESCRIPTION
**Pull request by SukkaW**
## What does it do?

The part of #3890

https://github.com/hexojs/hexo/issues/3890#issuecomment-573489000

## How to test

```sh
git clone -b feat-ignore-theme-in-plugin https://github.com/sukkaw/hexo.git
cd hexo
npm install
npm test
```

## Pull request tasks

- [x] Add test cases for the changes.
- [x] Passed the CI test.

### CODE REVIEW

File: test/scripts/hexo/load_plugins.js [hunk 1]
```diff
@@ -142,6 +159,20 @@ describe('Load plugins', () => {
     });
   });
 
+  it('ignore plugins which is typescript typescript definition', () => {
```
Comment (SukkaW) [inline test/scripts/hexo/load_plugins.js]: Improve test coverage.

File: lib/hexo/load_plugins.js [hunk 2]
```diff
@@ -12,25 +12,38 @@ module.exports = ctx => {
...
 function loadModuleList(ctx) {
-  if (ctx.config && Array.isArray(ctx.config.plugins)) {
-    return Promise.resolve(ctx.config.plugins).filter(item => typeof item === 'string');
+  let customThemeName;
+
+  if (ctx.config) {
+    const { theme, plugins } = ctx.config;
+
+    if (Array.isArray(plugins)) {
+      return Promise.resolve(ctx.config.plugins).filter(item => typeof item === 'string');
```
Comment (YoshinoriN) [inline lib/hexo/load_plugins.js]: `ctx.config.plugins` -> `plugins`

Comment (SukkaW) [reply lib/hexo/load_plugins.js]: I am not sure whether the context is required when return. I will have a test before changing it.

File: test/scripts/hexo/load_plugins.js [hunk 3]
```diff
@@ -128,6 +128,23 @@ describe('Load plugins', () => {
     });
   });
 
+  it('ignore plugin whose name is "hexo-theme-[hexo.config.theme]"', () => {
```
Comment (curbengh) [inline test/scripts/hexo/load_plugins.js]: suggest new tests to use async.

Comment (SukkaW) [reply test/scripts/hexo/load_plugins.js]: Change to async and rebased.

**Comment by YoshinoriN** [comment]
PS

The Appveyor build was failed. But, my local PC (Windows) test was passed.
I think no-problem of this error.

**Comment by SukkaW** [comment]
> The Appveyor build was failed. But, my local PC (Windows node 13.0.1) test was passed.

I believe the issue is related with Windows file system problem.

**Pull request merged**